### PR TITLE
inchi 1.07.1 (new formula); open-babel rdkit: use `inchi`

### PR DESCRIPTION
**Pull request by cho-m**
Audit will fail due to patch and notability will also fail. Copy of failure from CI before https://github.com/Homebrew/homebrew-core/labels/CI-skip-new-formulae
```
==> FAILED
Full audit --formula inchi --online --new output
  inchi
    * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
    * Formulae should not require patches to build. Patches should be submitted and accepted upstream first.
  Error: 2 problems in 1 formula detected.
```

Build system isn't the greatest. There is an upstream PR for CMake, but it lacks install targets so won't help there.

Mainly want to avoid building this inside 2 formulae.

Repology Ref: https://repology.org/project/inchi/versions

### CODE REVIEW

File: Formula/i/inchi.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  def install
+    bin.mkpath
+    lib.mkpath
+
+    args = ["C_COMPILER=#{ENV.cc}", "BIN_DIR=#{bin}", "LIB_DIR=#{lib}"]
+    system "make", "-C", "INCHI_API/libinchi/gcc", *args
+    system "make", "-C", "INCHI_EXE/inchi-1/gcc", *args
+
+    # Add major versioned and unversioned symlinks
+    libinchi = shared_library("libinchi", version.to_s[/^(\d+\.\d+)/, 1])
```
Comment (carlocab) [inline Formula/i/inchi.rb:24]:
```suggestion
    libinchi = shared_library("libinchi", version.major_minor)
```
Isn't this equivalent?

**Review by carlocab — APPROVED** [review]

File: Formula/i/inchi.rb (hunk 1 above)
Comment (cho-m) [reply Formula/i/inchi.rb:24]: It doesn't retain zeros so ends up with `1.7`.

Comment (carlocab) [reply Formula/i/inchi.rb:24]: Even if manually specify `version` in the formula? Huh.

Comment (cho-m) [reply Formula/i/inchi.rb:24]: It's mainly during the processing of the `version` to alternate forms:
```
brew(main):001> Formula["inchi"].version.to_s
=> "1.07.1"
brew(main):002> Formula["inchi"].version.major_minor.to_s
=> "1.7"
brew(main):003> Formula["inchi"].version.major_minor_patch.to_s
=> "1.7.1"
```

**Review by chenrui333 — APPROVED** [review]

File: Formula/i/inchi.rb (hunk 1 above)
Comment (carlocab) [reply Formula/i/inchi.rb:24]: Seems like a bug in `brew`.

**Review by carlocab — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Pull request merged**